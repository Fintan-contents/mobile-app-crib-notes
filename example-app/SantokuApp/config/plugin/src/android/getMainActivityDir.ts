import {AndroidConfig} from '@expo/config-plugins';
import path from 'path';

export async function getMainActivityDir(projectRoot: string) {
  const mainActivity = await AndroidConfig.Paths.getMainActivityAsync(projectRoot);
  return path.dirname(mainActivity.path);
}
