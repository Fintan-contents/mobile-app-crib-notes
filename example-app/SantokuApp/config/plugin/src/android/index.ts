export * from './withAndroidAddAppActivity';
export * from './withAndroidAddAppActivityAndroidManifest';
export * from './withAndroidAddLicensePlugin';
export * from './withAndroidAppBuildGradleForRelease';
export * from './withAndroidCopyTemplateJavaFile';
export * from './withAndroidRemoveUsesClearTextTrafficForRelease';
export * from './withAndroidDisabledWindowDrawsSystemBarBackgrounds';
export * from './withAndroidEnabledStatusBarTranslucent';
export * from './withAndroidCopyMainActivity';
export * from './withAndroidMoveDevSettingsActivityToDebugAndroidManifest';
export * from './withAndroidCopySplashScreenImages';
export * from './withAndroidCopyTemplateSplashScreenXml';
export * from './withAndroidFlexibleSplashScreen';
