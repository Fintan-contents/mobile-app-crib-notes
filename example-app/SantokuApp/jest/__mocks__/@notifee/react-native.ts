import notifee from '@notifee/react-native';
export * from '@notifee/react-native/dist/types/Library';
export * from '@notifee/react-native/dist/types/Notification';
export * from '@notifee/react-native/dist/types/Trigger';
export * from '@notifee/react-native/dist/types/NotificationIOS';
export * from '@notifee/react-native/dist/types/NotificationAndroid';

const mock: typeof notifee = {
  cancelAllNotifications: jest.fn(),
  cancelDisplayedNotifications: jest.fn(),
  cancelTriggerNotifications: jest.fn(),
  cancelNotification: jest.fn(),
  cancelDisplayedNotification: jest.fn(),
  cancelTriggerNotification: jest.fn(),
  createChannel: jest.fn(),
  createChannels: jest.fn(),
  createChannelGroup: jest.fn(),
  createChannelGroups: jest.fn(),
  deleteChannel: jest.fn(),
  deleteChannelGroup: jest.fn(),
  displayNotification: jest.fn(),
  openAlarmPermissionSettings: jest.fn(),
  createTriggerNotification: jest.fn(),
  getTriggerNotificationIds: jest.fn(),
  getDisplayedNotifications: jest.fn(),
  getTriggerNotifications: jest.fn(),
  getChannel: jest.fn(),
  isChannelCreated: jest.fn(),
  isChannelBlocked: jest.fn(),
  getChannels: jest.fn(),
  getChannelGroup: jest.fn(),
  getChannelGroups: jest.fn(),
  getInitialNotification: jest.fn(),
  onBackgroundEvent: jest.fn(),
  onForegroundEvent: jest.fn(),
  openNotificationSettings: jest.fn(),
  registerForegroundService: jest.fn(),
  stopForegroundService: jest.fn(),
  requestPermission: jest.fn(),
  setNotificationCategories: jest.fn(),
  getNotificationCategories: jest.fn(),
  getNotificationSettings: jest.fn(),
  getBadgeCount: jest.fn(),
  setBadgeCount: jest.fn(),
  incrementBadgeCount: jest.fn(),
  decrementBadgeCount: jest.fn(),
  openBatteryOptimizationSettings: jest.fn(),
  isBatteryOptimizationEnabled: jest.fn(),
  getPowerManagerInfo: jest.fn(),
  openPowerManagerSettings: jest.fn(),
  hideNotificationDrawer: jest.fn(),
  SDK_VERSION: '5.4.1',
};

// 複数のファイルでnotifeeをimportしていた場合に、redefineされないようにモックが存在するか確認
if (!__mocks.notifee) {
  Object.defineProperty(__mocks, 'notifee', {value: mock});
}

// テストケースごとにモックは初期化しておく。
beforeEach(() =>
  Object.values(mock).forEach(fn => {
    if (jest.isMockFunction(fn)) {
      fn.mockClear();
    }
  }),
);

export default mock;
