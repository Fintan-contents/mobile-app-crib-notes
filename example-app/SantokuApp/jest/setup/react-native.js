// https://reactnavigation.org/docs/testing#mocking-native-modules
// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// !!! NOTICE !!!
// The React Navigation documentation says that `react-native-reanimated` and `react-native-gesture-handler`
// also need to be configured with mocks.
// But since React Navigation v6 does not already depend on those packages, mocking those packages is not necessary
// for React Navigation v6.
//
// If you want to use `react-native-reanimated` and `react-native-gesture-handler`, add mock according to the guide
// provided by each package.
// - `react-native-reanimated`: https://docs.swmansion.com/react-native-reanimated/docs/guide/testing/
// - `react-native-gesture-handler`: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/testing
