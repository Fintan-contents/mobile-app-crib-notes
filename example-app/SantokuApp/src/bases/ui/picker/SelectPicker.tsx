import React from 'react';
import {Pressable, StyleSheet, TextInput, TextInputProps as RNETextInputProps, View, ViewProps} from 'react-native';

import {DefaultPickerAccessory, DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdrop, PickerBackdropProps} from './PickerBackdrop';
import {PickerContainer, PickerContainerProps} from './PickerContainer';
import {SelectPickerItems, SelectPickerItemsProps} from './SelectPickerItems';
import {useSelectPicker} from './useSelectPicker';

export type Item<T> = {
  label: string;
  value: T;
  inputLabel?: string;
  key?: React.Key;
  color?: string;
  fontFamily?: string;
};

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;
export type SelectPickerProps<ItemT> = {
  /**
   * SelectPickerに表示するアイテム
   */
  items: Item<ItemT>[];
  /**
   * 選択されたアイテムのKey
   */
  selectedItemKey?: React.Key | ItemT;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (selectedItem?: Item<ItemT>) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   */
  onDismiss?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDelete?: (selectedItem?: Item<ItemT>) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onCancel?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  onDone?: (selectedItem?: Item<ItemT>) => void;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、{@link DefaultPickerAccessory}がデフォルトで表示されます。
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、{@link SelectPickerItems}がデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * Keyを作成する関数
   */
  keyExtractor?: ((item: Item<ItemT>, index: number) => string) | undefined;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   */
  pickerProps?: Omit<SelectPickerItemsProps<ItemT>, 'items' | 'selectedValue' | 'onValueChange' | 'keyExtractor'>;
  /**
   * PickerBackdropのProps
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

const DEFAULT_DURATION = 300;

export const SelectPicker = <ItemT extends unknown>(props: SelectPickerProps<ItemT>) => {
  const {isVisible, inputValue, handleBackdropPress, onValueChange, open, handleDelete, handleCancel, handleDone} =
    useSelectPicker<ItemT>(props);
  const {
    items,
    selectedItemKey,
    placeholder,
    textInputProps,
    keyExtractor,
    pickerAccessoryComponent,
    pickerItemsComponent,
    textInputComponent,
    pickerItemsContainerProps: {style: pickerItemsContainerStyle, ...pickerItemsContainerProps} = {},
    pickerProps,
    pickerBackdropProps: {
      fadeInDuration = DEFAULT_DURATION,
      fadeOutDuration = DEFAULT_DURATION,
      ...pickerBackdropProps
    } = {},
    pickerContainerProps: {
      style: pickerContainerStyle,
      slideInDuration = DEFAULT_DURATION,
      slideOutDuration = DEFAULT_DURATION,
      ...pickerContainerProps
    } = {},
    pickerAccessoryProps,
  } = props;

  return (
    <>
      <PickerBackdrop
        isVisible={isVisible}
        onPress={handleBackdropPress}
        fadeInDuration={fadeInDuration}
        fadeOutDuration={fadeOutDuration}
        {...pickerBackdropProps}>
        <PickerContainer
          isVisible={isVisible}
          style={[styles.pickerContainer, pickerContainerStyle]}
          slideInDuration={slideInDuration}
          slideOutDuration={slideOutDuration}
          {...pickerContainerProps}>
          {pickerAccessoryComponent ? (
            pickerAccessoryComponent
          ) : (
            <DefaultPickerAccessory
              onDelete={handleDelete}
              onCancel={handleCancel}
              onDone={handleDone}
              {...pickerAccessoryProps}
            />
          )}
          <View {...pickerItemsContainerProps}>
            {pickerItemsComponent ? (
              pickerItemsComponent
            ) : (
              <SelectPickerItems
                selectedValue={selectedItemKey}
                items={items}
                onValueChange={onValueChange}
                keyExtractor={keyExtractor}
                {...pickerProps}
              />
            )}
          </View>
        </PickerContainer>
      </PickerBackdrop>
      <Pressable onPress={open} testID="pressableContainer">
        <View pointerEvents="box-only">
          {textInputComponent ? (
            textInputComponent
          ) : (
            // テキスト入力とスタイルを合わせるために、TextではなくTextInputを使用する
            <TextInput placeholder={placeholder} value={inputValue} editable={false} {...textInputProps} />
          )}
        </View>
      </Pressable>
    </>
  );
};

const styles = StyleSheet.create({
  pickerContainer: {
    backgroundColor: 'white',
  },
});
