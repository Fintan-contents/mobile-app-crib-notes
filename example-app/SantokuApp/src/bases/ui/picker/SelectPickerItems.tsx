import {PickerProps} from '@react-native-picker/picker';
import {log} from 'bases/logging';
import React from 'react';
import {Platform, StyleProp, TextStyle} from 'react-native';

import {Item} from './SelectPicker';

export type SelectPickerItemsProps<ItemT> = Omit<
  PickerProps<React.Key | ItemT>,
  'enabled' | 'mode' | 'prompt' | 'dropdownIconColor' | 'dropdownIconRippleColor' | 'onFocus' | 'onBlur' | 'style'
> & {
  items: Item<ItemT>[];
  keyExtractor?: ((item: Item<ItemT>, index: number) => string) | undefined;
  /**
   * Pickerのスタイル
   * @platform ios
   */
  style?: StyleProp<TextStyle>;
  /**
   * アイテムの高さ
   * @platform android
   */
  itemHeight?: number;
  /**
   * 選択状態のアイテムのテキストカラー
   * @platform android
   */
  activeColor?: string;
  /**
   * 未選択状態のアイテムのテキストカラー
   * @platform android
   */
  inactiveColor?: string;
  /**
   * FlatListのtestID
   * @platform android
   */
  flatListTestID?: string;
  /**
   * SeparatorのtestID
   * @platform android
   */
  separatorTestID?: string;
  /**
   * SelectPickerItemのPressableのtestIDプレフィックス
   * @platform android
   */
  itemPressableTestIDPrefix?: string;
  /**
   * SelectPickerItemのTextのtestIDプレフィックス
   * @platform android
   */
  itemTextTestIDPrefix?: string;
};

export const SelectPickerItems = <ItemT extends unknown>(props: SelectPickerItemsProps<ItemT>) => {
  React.useEffect(() => {
    log.warn(`SelectPickerItems is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
