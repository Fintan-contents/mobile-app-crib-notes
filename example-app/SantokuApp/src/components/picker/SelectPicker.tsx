import {PickerProps} from '@react-native-picker/picker';
import {log} from 'framework/logging';
import React from 'react';
import {Platform, StyleProp, TextStyle, TextInputProps as RNETextInputProps, ViewProps} from 'react-native';

import {DefaultPickerAccessoryProps} from './DefaultPickerAccessory';
import {PickerBackdropProps} from './PickerBackdrop';
import {PickerContainerProps} from './PickerContainer';

export type Item<T> = {
  label: string;
  value: T;
  inputLabel?: string;
  key?: React.Key;
  color?: string;
  fontFamily?: string;
  /**
   * アイテムに指定するスタイル
   * 以下のプロパティのみ使用できます。
   * - 'color'
   * - 'backgroundColor'
   * - 'fontSize'
   * - 'fontFamily'
   *
   * @platform android
   */
  style?: StyleProp<Pick<TextStyle, 'color' | 'backgroundColor' | 'fontSize' | 'fontFamily'>>;
};

type TextInputProps = Omit<RNETextInputProps, 'value' | 'editable'>;
export type SelectPickerProps<ItemT> = {
  /**
   * SelectPickerに表示するアイテム
   */
  items: Item<ItemT>[];
  /**
   * 選択されたアイテムのKey
   */
  selectedItemKey?: React.Key;
  /**
   * アイテムが選択された場合に呼び出される関数
   */
  onSelectedItemChange?: (selectedItem?: Item<ItemT>) => void;
  /**
   * PickerBackdropをタップして閉じた場合に呼び出される関数
   *
   * @platform ios
   */
  onDismiss?: (selectedItem?: Item<ItemT>) => void;
  /**
   * DeleteLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * @platform ios
   */
  onDelete?: (selectedItem?: Item<ItemT>) => void;
  /**
   * CancelLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * @platform ios
   */
  onCancel?: (selectedItem?: Item<ItemT>) => void;
  /**
   * [iOS]
   * DoneLabelがタップされた場合に呼び出される関数
   * タップ後、SelectPickerは自動で閉じます。
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * [Android]
   * アイテムが選択された場合に呼び出される関数
   */
  onDone?: (selectedItem?: Item<ItemT>) => void;
  /**
   * プレースホルダ
   */
  placeholder?: string;
  /**
   * ヘッダコンポーネント
   * 指定しない場合は、PickerAccessoryがデフォルトで表示されます。
   *
   * @platform ios
   */
  pickerAccessoryComponent?: React.ReactNode;
  /**
   * Pickerコンポーネント
   * 指定しない場合は、SelectPickerItemsがデフォルトで表示されます。
   */
  pickerItemsComponent?: React.ReactNode;
  /**
   * 選択されたアイテムを表示するテキストコンポーネント
   */
  textInputComponent?: React.ReactNode;
  /**
   * Keyを作成する関数
   */
  keyExtractor?: ((item: ItemT, index: number) => React.Key) | undefined;
  /**
   * 選択されたアイテムを表示するテキストコンポーネントのProps
   * なお、textInputComponentを指定した場合は使用されません。
   */
  textInputProps?: TextInputProps;
  /**
   * Pickerコンポーネントを囲むContainerのProps
   *
   * @platform ios
   */
  pickerItemsContainerProps?: ViewProps;
  /**
   * PickerのProps
   * なお、pickerItemsComponentを指定した場合は使用されません。
   */
  pickerProps?: Omit<PickerProps<ItemT>, 'selectedValue' | 'onValueChange'>;
  /**
   * PickerBackdropのProps
   *
   * @platform ios
   */
  pickerBackdropProps?: Omit<PickerBackdropProps, 'isVisible' | 'onPress'>;
  /**
   * PickerContainerのProps
   *
   * @platform ios
   */
  pickerContainerProps?: Omit<PickerContainerProps, 'isVisible'>;
  /**
   * PickerAccessoryのProps
   * なお、pickerAccessoryComponentを指定した場合は使用されません。
   *
   * @platform ios
   */
  pickerAccessoryProps?: Omit<DefaultPickerAccessoryProps, 'onDelete' | 'onCancel' | 'onDone'>;
};

export const SelectPicker = <ItemT extends unknown>(props: SelectPickerProps<ItemT>) => {
  React.useEffect(() => {
    log.warn(`SelectPicker is not supported on: ${Platform.OS}`);
  }, []);
  return null;
};
