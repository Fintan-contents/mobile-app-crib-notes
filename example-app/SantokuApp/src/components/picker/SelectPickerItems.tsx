import {Picker, PickerProps} from '@react-native-picker/picker';
import React from 'react';

import {Item} from './SelectPicker';

export type SelectPickerItemsProps<ItemT> = PickerProps<React.Key> & {
  items: Item<ItemT>[];
  keyExtractor?: ((item: ItemT, index: number) => React.Key) | undefined;
};

export const SelectPickerItems = <ItemT extends unknown>({
  selectedValue,
  onValueChange,
  items,
  keyExtractor,
  ...rest
}: SelectPickerItemsProps<ItemT>) => {
  return (
    <Picker selectedValue={selectedValue} onValueChange={onValueChange} {...rest}>
      {items.map((item, index) => (
        <Picker.Item
          key={keyExtractor ? keyExtractor(item.value, index) : item.key ?? item.label}
          label={item.label}
          value={item.key ?? item.value}
          color={item.color}
          fontFamily={item.fontFamily}
          style={item.style}
        />
      ))}
    </Picker>
  );
};
