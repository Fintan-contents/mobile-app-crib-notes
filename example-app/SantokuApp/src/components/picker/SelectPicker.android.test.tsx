import {render} from '@testing-library/react-native';
import React from 'react';
import {TextInputProps, View} from 'react-native';

import {SelectPicker} from './SelectPicker.android';
import {SelectPickerItemsProps} from './SelectPickerItems';

describe('SelectPicker only with required props', () => {
  it('renders successfully', () => {
    const items = [
      {value: '1', label: 'test1'},
      {value: '2', label: 'test2'},
    ];
    const sut = render(
      <SelectPicker
        items={items}
        selectedItemKey="1"
        textInputProps={{testID: 'textInput'}}
        pickerProps={{testID: 'picker'}}
      />,
    );

    expect(sut).toMatchSnapshot('SelectPicker  only with required props.');

    const picker = sut.queryByTestId('picker');
    const textInput = sut.queryByTestId('textInput');
    expect(picker).not.toBeNull();
    expect(textInput).not.toBeNull();
  });
});

describe('SelectPicker with all props', () => {
  it('should be applied all properly with default xxx component', () => {
    const items = [
      {value: '1', label: 'test1'},
      {value: '2', label: 'test2'},
    ];
    const onSelectedItemChange = jest.fn();
    const onDone = jest.fn();
    const keyExtractor = jest.fn((value: string, index: number) => `${value}-${index}`);
    /**
     * onDoneは自動テストではイベント発火できませんでした
     */
    const sut = render(
      <SelectPicker
        items={items}
        selectedItemKey="1"
        onSelectedItemChange={onSelectedItemChange}
        placeholder="please select..."
        keyExtractor={keyExtractor}
        textInputProps={{style: {color: 'red'}, testID: 'textInput'}}
        pickerProps={{numberOfLines: 5, testID: 'picker'}}
        onDone={onDone}
      />,
    );
    expect(sut).toMatchSnapshot('SelectPicker all properly with default xxx component.');

    // assert PickerItems
    const picker = sut.getByTestId('picker');
    const pickerProps = picker.props as SelectPickerItemsProps<string>;
    expect(pickerProps.numberOfLines).toBe(5);
    expect(pickerProps.items.length).toBe(2);

    // assert textInput
    const textInput = sut.getByTestId('textInput');
    const textInputProps = textInput.props as TextInputProps;
    expect(textInputProps.style).toEqual({color: 'red'});
    expect(keyExtractor).toHaveBeenNthCalledWith(1, '1', 0);
    expect(keyExtractor).toHaveBeenNthCalledWith(2, '2', 1);
  });

  it('should be applied all properly with custom xxx component', () => {
    const CustomTextInput = <View testID="customTextInput" />;
    const items = [
      {value: '1', label: 'test1'},
      {value: '2', label: 'test2'},
    ];
    const onSelectedItemChange = jest.fn();
    const sut = render(
      <SelectPicker
        items={items}
        selectedItemKey="1"
        onSelectedItemChange={onSelectedItemChange}
        placeholder="please select..."
        textInputProps={{testID: 'defaultTextInput'}}
        pickerProps={{numberOfLines: 5, testID: 'picker'}}
        textInputComponent={CustomTextInput}
      />,
    );
    expect(sut).toMatchSnapshot('SelectPicker all properly with custom xxx component.');

    // assert PickerItems
    const picker = sut.getByTestId('picker');
    const pickerProps = picker.props as SelectPickerItemsProps<string>;
    expect(pickerProps.numberOfLines).toBe(5);
    expect(pickerProps.items.length).toBe(2);

    // assert textInput
    const defaultTextInput = sut.queryByTestId('defaultTextInput');
    const customTextInput = sut.queryByTestId('customTextInput');
    expect(defaultTextInput).toBeNull();
    expect(customTextInput).not.toBeNull();
  });
});
