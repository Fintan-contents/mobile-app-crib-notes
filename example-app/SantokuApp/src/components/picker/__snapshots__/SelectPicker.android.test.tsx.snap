// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectPicker only with required props renders successfully: SelectPicker  only with required props. 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    editable={false}
    rejectResponderTermination={true}
    testID="textInput"
    underlineColorAndroid="transparent"
    value="test1"
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "transparent",
        "height": "100%",
        "opacity": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "test1",
            "testID": undefined,
            "textColor": undefined,
            "value": "1",
          },
          Object {
            "label": "test2",
            "testID": undefined,
            "textColor": undefined,
            "value": "2",
          },
        ]
      }
      numberOfLines={1}
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="picker"
    />
  </View>
</View>
`;

exports[`SelectPicker with all props should be applied all properly with custom xxx component: SelectPicker all properly with custom xxx component. 1`] = `
<View>
  <View
    testID="customTextInput"
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "transparent",
        "height": "100%",
        "opacity": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "test1",
            "testID": undefined,
            "textColor": undefined,
            "value": "1",
          },
          Object {
            "label": "test2",
            "testID": undefined,
            "textColor": undefined,
            "value": "2",
          },
        ]
      }
      numberOfLines={5}
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="picker"
    />
  </View>
</View>
`;

exports[`SelectPicker with all props should be applied all properly with default xxx component: SelectPicker all properly with default xxx component. 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    editable={false}
    placeholder="please select..."
    rejectResponderTermination={true}
    style={
      Object {
        "color": "red",
      }
    }
    testID="textInput"
    underlineColorAndroid="transparent"
    value="test1"
  />
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "transparent",
        "height": "100%",
        "opacity": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNCPicker
      items={
        Array [
          Object {
            "label": "test1",
            "testID": undefined,
            "textColor": undefined,
            "value": "1",
          },
          Object {
            "label": "test2",
            "testID": undefined,
            "textColor": undefined,
            "value": "2",
          },
        ]
      }
      numberOfLines={5}
      onChange={[Function]}
      selectedIndex={0}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
      testID="picker"
    />
  </View>
</View>
`;
